tests/test-data/projects/no-docker-compose/basic/src/windows-crlf.sh

```diff
@@ -1,2 +1,2 @@
-#!/bin/sh
-echo 'this file was saved with Windows CRLF line endings'
+#!/bin/sh
+echo 'this file was saved with Windows CRLF line endings'
```

Comment:
@Page-, I've had to re-add this file (unmodified) in order to avoid `git status` reporting modifications (and preventing branch switching, pushing...) despite not touching the file:
* Merely cloning the CLI master branch (fresh, on a new folder) and running `git status` would report that the file was modified.
* I tried with different values of `autocrlf` in my `~/.giconfig` file. No difference.
* Both before and after this commit:
  * `ls -l` reported the file size unchanged (70 bytes).
  * `hexdump -C` showed it contained CRLF as expected.
  * `md5` showed the same hash (1fffe84f62b4609d481dc0c00ca0887b).
* We already had test cases checking the file size before and after the CLI performs EOL conversion -- the tests verify that the file size is 70 before conversion, and 68 after conversion (2 lines). The tests passed before and after this commit.

The git-scm documentation for the `.gitattributes` file (which we recently added to the repo) seems to warn about this potential condition, with the fix being to re-add the file:
> https://git-scm.com/docs/gitattributes#_code_eol_code
> **eol**
> This attribute sets a specific line-ending style to be used in the working directory. It enables end-of-line conversion without any content checks, effectively setting the text attribute. **Note that setting this attribute on paths which are in the index with CRLF line endings may make the paths to be considered dirty. Adding the path to the index again will normalize the line endings in the index.**

_"Adding the path to the index again will normalize the line endings in the index"_ -- this is thus what I am trying to achieve with this commit.

Reply:
Also: `git checkout `, `git rm --cached -r .`, `git reset --hard` ... make not difference. Well, re-cloning the CLI on a new folder makes no difference...
```
$ git checkout -- tests/test-data/projects/no-docker-compose/basic/src/windows-crlf.sh

$ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   tests/test-data/projects/no-docker-compose/basic/src/windows-crlf.sh

no changes added to commit (use "git add" and/or "git commit -a")

$ git reset --hard master
HEAD is now at d48672f v11.25.17

$ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   tests/test-data/projects/no-docker-compose/basic/src/windows-crlf.sh

no changes added to commit (use "git add" and/or "git commit -a")

$ git checkout 1611-replace-mmmagic
error: Your local changes to the following files would be overwritten by checkout:
	tests/test-data/projects/no-docker-compose/basic/src/windows-crlf.sh
Please commit your changes or stash them before you switch branches.
Aborting
```

Reply:
Steps to reproduce:
```
$ docker run -it ubuntu:18.04
$ apt-get update && apt-get install -y git
$ git clone https://github.com/balena-io/balena-cli.git
$ cd balena-cli
$ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   tests/test-data/projects/no-docker-compose/basic/src/windows-crlf.sh

no changes added to commit (use "git add" and/or "git commit -a")
```
Steps to fix, following on from the above:
```
# set username and email, so that git allows creating a commit
$ git config --global user.name "test"
$ git config --global user.email "test"
$ git add tests/test-data/projects/no-docker-compose/basic/src/windows-crlf.sh
$ git commit -am "fixed"
$ git status
On branch master
Your branch is ahead of 'origin/master' by 1 commit.
  (use "git push" to publish your local commits)

nothing to commit, working tree clean
```